caddyhttp/httpserver/server.go

```diff
@@ -126,15 +126,16 @@ func (s *Server) Listen() (net.Listener, error) {
 		ln = tcpKeepAliveListener{TCPListener: tcpLn}
 	}
 
+	cln := ln.(caddy.Listener)
```

Comment:
Is this type assertion really needed? A caddy.Listener is also a net.Listener, and there's a type assertion at the return statement.